### ISSUE TITLE
Setup of AlarmR

### ISSUE THREAD

**Issue by pzofio** [sqall01/alertR#12]
Hi sqall, I've been looking into your project, but after many tries I could not make it work. I found some typos in the example configuration on the managerclientdatabase section and could not make the web mobile manager to work (missing config.php)

Is it possible to connect to the system via a regular web client? How can I setup it correctly? 
Your project seems very interesting, but I could not find enough documentation on these parts .

Please point me in the right direction.

Pau

**Comment by sqall01** [comment]
Hi Pau,

did you take a look into the wiki of the project? More specifically, this here:

https://github.com/sqall01/alertR/wiki/Example-Configuration#alertr_manager_client_database

If not, please do so. At the moment I have no idea what you did and what you are trying to do. Can you please specify your problem?

Can you please tell me what typos you found? Without it I can not fix them.

Why is the config.php missing? I just checked, the file config.php.template exists. Did you renamed it to config.php if you were using it as a template? A example configuration for it is also given here:

https://github.com/sqall01/alertR/wiki/Example-Configuration#alertr_web_mobile_manager

Furthermore, what do you mean with "connect to the system via a regular web client"? What is a "regular web client"?

Cheers,

sqall

**Comment by pzofio** [comment]
Hi Andre, thank you for your answer.Your project looks amazing and I want to implement it and play with it so I learn in the process. I'm a mechanical engineer but only have basic programming skills and basic Unix/linux user skill.
The system i want to install will have two raspberrys and I have wired one Input  and one output both in the server and the client, I have already checked that they work by directly working on the GPIOs.
I followed your configuration example, in order to setup first the server. I could not understand clearly what packages where essential so I follow the guide andconfigured all of them except:
 alertR Manager Client KeypadalertR Alert Client XBMCalertR Sensor Client FIFOIt appears that all the services are running, and I configured the GPIO of the door security to check if I receved an email when I push the button I have wired, but did not work.
Problems with the example configuration
On the example configuration at the beginning you use user pi, but in the middle you use sqall as user.
Also I found a typo in the alertR Manger Client Databasewhich is not the title:
 first you make a directory
 sqall@webserver:/home/sqall/alertR# mkdir managerClientDatabase
but in the config file your example configuration says:
DAEMON=/home/sqall/alertR/alertRmanagerDatabase/alertRclient.pyI understand that if we are in the section configuring the manager client database it should point to managerClientDatabase
Anyway I could not install the android app and I tried to connect via web browser (google chrome pointing to the alertR\index.php) but it did not work.
Also looking at the configuration of the web page i found:
$configUnixSocketPath = "/home/sqall/alertR/alertRmanagerDatabase/config/localsocket";
I could not find localsocket, only config.conf and the template file. I don't know if it was because of this or bad configuration of the apache server, but it did not work.
Questions
How should I install the android client on my mobile?How are the alert emails sent? Should I configure first any smtp server?Can I check/configure the whole system from a webpage in Google Chrome?
Anyway congratulations on your project, maybe I lack the programming skills to understand or make it work, but sure is much more interesting than the other similar projects I've  been testing (although the installation guides may be simpler/ more noob friendly)
Best Regards,
Pau




Date: Mon, 28 Mar 2016 09:13:52 -0700

**Comment by sqall01** [comment]
Ah I see. The "bad" tutorial is definitely a problem and the next thing on my agenda after finishing the next version (I hope it will be finished in 1-2 months). But I will fix the typos.

The "basic configuration" switches between users because they are all not dependent on each other. Meaning you have installed a lot of clients that you do not need. The web service is of course reachable via a normal browser like Chrome. And for the eMails you need a local eMail service like postfix on your host installed. But let us do one thing at a time and build up from there.

I would suggest we first test if the GPIO setup is working and then setup the server with two clients (one sensor client for the GPIOs and one graphical client that we can see if everything works).

Did you test if your hardware setup with the GPIOs works? I would guess that you added your door via a normal magnetic reed switch with a circuit like this:

https://github.com/sqall01/alertR/blob/master/docs/raspberry_pi_pull_up_for_switch.jpg

Is this correct? If so, can you please try to use this script here:

https://github.com/sqall01/alertR/blob/master/helperScripts/raspberryPiGpioInputTest.py

to test if everything is working? This script will print a 1 or a 0 depending on the state of the switch. The only thing you have to do in this script is configuring the correct GPIO pin number.

Cheers,

sqall

**Comment by pzofio** [comment]
Hi Andre, the tutorial is not bad, just too short IMHO. It would be helpful to have  two or three configurations a very basic one to set up the server with just one rasppi and mobile/web control of the server, another one  with one server and one client (sensor), and the one you have where there are mutiple clients of many types.
Anyway I already used  the helperscripts and set up the GPIOs according to the circuit you tell me. I tested them and configured  the client sensor so when I push the input switch simulating the magnetic reed switch sends me an email, but now I know that I have to configure postfix in order to the emails to be sent. 
You are working on your new version, I don't want to take too much time of yours so I'll be waiting for your new version and give it a new try.
Thank you very much and kudos on your project.
Regards,
Pau

Date: Tue, 29 Mar 2016 09:04:39 -0700
